Remove Entity and update some SideBar related classes

ghost91- opened the pull request:
Closes #431
Closes #370
Closes #362
Closes #419
Closes #427
Closes #363
Closes #425

UFOMelkor commented on src/foundry/foundry.js/applications/compendium.d.ts:
```diff
@@ -1,297 +1,85 @@
+  static get defaultOptions(): Application.Options;
 
   /** @override */
   getTitle(): string;
```
```suggestion
  get title(): string;
```

UFOMelkor commented on src/foundry/foundry.js/applications/compendium.d.ts:
```diff
@@ -1,297 +1,85 @@
-   */
-  protected _onDrop(event: DragEvent): Promise<false | Entity>;
+  /** @override */
+  protected _onDrop(event: DragEvent): ReturnType<this['collection']['importDocument']>;
```
```suggestion
  protected _onDrop(event: DragEvent): ReturnType<this['collection']['importDocument']> | false;
```

UFOMelkor commented on src/foundry/foundry.js/applications/formApplications/imagePopout.d.ts:
```diff
@@ -49,6 +54,7 @@ declare class ImagePopout extends FormApplication<ImagePopout.Options, ImagePopo
    * Determine the correct position and dimensions for the displayed image
+   * @returns The positioning object which should be used for rendering
    */
   protected static getPosition(img: string): Application.Position;
```
```suggestion
  protected static getPosition(img: string): Promise<{ width: number; height: number } | { width: number; height: number; top: number; left: number }>;
```

UFOMelkor commented on src/foundry/foundry.js/applications/formApplications/imagePopout.d.ts:
```diff
@@ -49,6 +54,7 @@ declare class ImagePopout extends FormApplication<ImagePopout.Options, ImagePopo
+   * @returns The positioning object which should be used for rendering
    */
   protected static getPosition(img: string): Application.Position;
 
```
- `_handleShareImage`: First parameter is optional.

UFOMelkor replied:
`_handleShareImage` (below): parameter is optional.

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTab.d.ts:
```diff
@@ -1,52 +1,64 @@
+  constructor(...args: ConstructorParameters<typeof Application>);
   /**
    * The base name of this sidebar tab
    */
```
```suggestion
   * @defaultValue `''`
   */
```

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTab.d.ts:
```diff
@@ -58,22 +70,3 @@ declare abstract class SidebarTab<P extends SidebarTab.Options = SidebarTab.Opti
    */
```
```suggestion
   * @param entries  - The entries which are now observed
   * @param observer - The intersection observer instance
   */
```

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/settings.d.ts:
```diff
@@ -1,12 +1,21 @@
+  static get defaultOptions(): Application.Options;
 
   /**
    * @param options - (unused)
```
`_onUpdateNotificationClick` (last method of the class) is missing

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/settings.d.ts:
```diff
@@ -35,21 +44,4 @@ declare namespace Settings {
     coreUpdate: string | false;
```
- `user` should be the configured document class.
- `isDemo` is missing
- `systemUpdate` is missing

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/sidebarDirectories/rollTableDirectory.d.ts:
```diff
@@ -1,40 +1,12 @@
+declare class RollTableDirectory<
+  Options extends SidebarDirectory.Options = SidebarDirectory.Options
+> extends SidebarDirectory<'RollTable', Options> {
+  static get defaultOptions(): SidebarDirectory.Options;
```
```suggestion
```

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/sidebarDirectory.d.ts:
```diff
@@ -1,261 +1,246 @@
-   * The named entity which this Sidebar Directory contains
-   * @remarks This method is abstract in SidebarTab.
+   * A shared pattern for the sidebar directory which Actors, Items, and Scenes all use
+   * @typeParam Options - The type of the options object
```
`@typeParam` for `Name` is missing

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/sidebarDirectory.d.ts:
```diff
@@ -1,261 +1,246 @@
+    /**
+     * Initialize the content of the directory by categorizing folders and entities into a hierarchical tree structure.
+     */
+    initialize(): void;
```
Doubled method?

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/sidebarDirectory.d.ts:
```diff
@@ -1,261 +1,246 @@
+     * @override
      */
-    id: string;
+    protected _onDrop(event: DragEvent): void;
```
```suggestion
    protected _onDrop(event: DragEvent): false | void | Promise<InstanceType<ConfiguredDocumentClassForName<Name>> | undefined> | Promise<InstanceType<ConfiguredDocumentClass<typeof Folder>> | undefined>;
```

UFOMelkor commented on src/foundry/foundry.js/applications/sidebarTabs/sidebarDirectory.d.ts:
```diff
@@ -1,261 +1,246 @@
-  interface Tree {
-    root: boolean;
+    interface Options extends Application.Options {
+      renderUpdateKeys: string[];
```
```suggestion
      renderUpdateKeys: string[];
      contextMenuSelector: string;
```

UFOMelkor reviewed (changes requested):
Looks good to me, just some small things I noticed.

ghost91- commented:
@UFOMelkor Addressed all of your comments. I hadn't actually planned to update `Settings`, I only made it compile and cleaned up a bit. But seeing from your comments that there wasn't much that was missing, I also addressed those comments to update `Settings` and added it to the list of issues that will be closed when this is merged.
